traversal/procedure/ProcedureEdge.java

```diff
@@ -1215,46 +1229,45 @@ public Forwardable<KeyValue<ThingVertex, ThingVertex>, Order.Asc> branchEdge(
                     ) {
                         assert fromVertex.isThing() && !roleTypes.isEmpty();
                         ThingVertex rel = fromVertex.asThing();
-                        Forwardable<KeyValue<ThingVertex, ThingVertex>, Order.Asc> iter;
 
                         Set<TypeVertex> relationRoleTypes = graphMgr.schema().relatedRoleTypes(rel.type());
                         FunctionalIterator<TypeVertex> instanceRoleTypes = iterate(relationRoleTypes)
                                 .filter(rt -> this.roleTypes.contains(rt.properLabel()));
-                        if (to.props().hasIID()) iter = branchToIID(graphMgr, params, rel, instanceRoleTypes);
-                        else iter = branchToTypes(graphMgr, rel, instanceRoleTypes);
-
-                        if (!to.props().predicates().isEmpty()) iter = to.filterPredicatesOnEdge(iter, params);
-                        return iter;
+                        if (to.props().hasIID()) {
+                            return to.iterateAndFilterPredicatesOnEdges(branchToIID(graphMgr, params, rel, instanceRoleTypes), params, ASC);
+                        } else {
+                            return to.mergeAndFilterPredicatesOnEdges(graphMgr, branchToTypes(graphMgr, rel, instanceRoleTypes), params, ASC);
```

Comment:
we represent optimised `Edge`s as a `KeyValue<ThingVertex,ThingVertex>`, where the key is the `to` of the edge and the value is the optimised role vertex. We have to rename the method with the suffix `...OnEdges` to differentiate versus the one with `...onVertices` which is overloaded & differentiated only with generics, which Java doesn't allow